cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -663,7 +739,75 @@ class row_hasher;
 
 namespace equality {
 
-template <typename Nullate>
+/**
+ * @brief Equality comparator functor that evaluates `NaN` not equal to all other values for IEE 754
+ * compliance.
+ */
+struct physical_equality_comparator {
+  /**
+   * @brief Shared operator for all `Element` types to check if `lhs` is equivalent to `rhs`. `nan
+   * != nan`.
+   *
+   * @param lhs first element
+   * @param rhs second element
+   * @return `true` if `lhs` == `rhs` else `false`.
+   */
+  template <typename Element>
+  __device__ bool operator()(Element const lhs, Element const rhs)
+  {
+    return lhs == rhs;
+  }
+};
+
+/**
+ * @brief Equality comparator functor that evaluates `NaN` as equivalent to other `NaN`s .
+ */
+struct sorting_physical_equality_comparator : physical_equality_comparator {
```

Comment:
`sorting_` looks like a copy/paste remnant given this is an equality comparator.

Reply:
`sorting_` was how I was differentiating 754 compliant and non-compliant functors. How does `nan_equal_physical_equality_comparator` sound?